openfisca_france/model/prelevements_obligatoires/impot_revenu/ir.py

```diff
@@ -2003,17 +2047,19 @@ def function(self, simulation, period):
         mncn_pvct_holder = simulation.compute('mncn_pvct', period)
         cncn_aimp_holder = simulation.compute('cncn_aimp', period)
         cncn_bene_holder = simulation.compute('cncn_bene', period)
-        cga = simulation.legislation_at(period.start).ir.rpns.cga_taux2
-        spbnc = simulation.legislation_at(period.start).ir.rpns.microentreprise.specialbnc
+        cga = simulation.legislation_at(period.start).impot_revenu.rpns.cga_taux2
+        specialbnc = simulation.legislation_at(period.start).impot_revenu.rpns.micro.specialbnc
 
         def abat_rpns(rev, P):
             return max_(0, rev - min_(rev, max_(P.taux * min_(P.max, rev), P.min)))
         cncn_bene = self.sum_by_entity(cncn_bene_holder)
         mncn_impo = self.sum_by_entity(mncn_impo_holder)
         mncn_pvct = self.sum_by_entity(mncn_pvct_holder)
         cncn_aimp = self.sum_by_entity(cncn_aimp_holder)
-        return period, min_(f5ht + f5it + f5jt + f5kt + f5lt + f5mt, abat_rpns(mncn_impo, spbnc) +
-                    mncn_pvct + cncn_aimp + (1 + cga) * cncn_bene)
+        return period, min_(
+            f5ht + f5it + f5jt + f5kt + f5lt + f5mt,
+            abat_rpns(mncn_impo, specialbnc.services) + mncn_pvct + cncn_aimp + (1 + cga) * cncn_bene
+            )  # TODO check !
```

Comment:
This `#TODO` is likely to be forgotten and to stay here forever ;).
You can either: 
    - Check it before merging and remove it
    - Add an issue about what we need to check, remove it, and maybe add a comment like "This has not been checked. See https://github.com/openfisca/openfisca-france/issue/xxx.